samples: tensorflow: fix cmakelists.txt files

laurenmurphyx64 opened the pull request:
Adds no-threadsafe-statics to compiler flags per [discussion](https://github.com/zephyrproject-rtos/zephyr/pull/34057#pullrequestreview-647564622) in #34057 and updates documentation to remove note that implied the flag is required to compile the module. The flag is only required to compile the samples. Changes from explicit file list in CMakeLists.txt to GLOB.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
samples: tensorflow: add no-threadsafe-statics to compiler flags

Adds no-threadsafe-statics to compiler flags for gcc and arcmwdt.

The flag is required to compile the samples - the module can compile
without it.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
samples: tensorflow: fix documentation

Fixes documentation for samples that incorrectly state that TensorFlow
library requires the -fno-threadsafe-statics flag to compile.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

evgeniy-paltsev commented on cmake/compiler/gcc/compiler_flags.cmake:
```diff
@@ -177,6 +177,9 @@ set_compiler_property(PROPERTY sanitize_address -fsanitize=address)
 set_compiler_property(PROPERTY sanitize_undefined -fsanitize=undefined)
 
+# GCC compiler flag for turning off thread-safe initialization of local statics
+set_property(TARGET compiler-cpp PROPERTY no_threadsafe_statics "-fno-threadsafe-statics")
```
May I ask you to add the same property configuration for MWDT toolchain?
MWDT supports `-fno-threadsafe-statics` flag, so it would be just a copy of this line
```
set_property(TARGET compiler-cpp PROPERTY no_threadsafe_statics "-fno-threadsafe-statics")
```
 to `cmake/compiler/arcmwdt/compiler_flags.cmake` file.

Thanks!

laurenmurphyx64 replied:
Added it to the file on the first commit, thank you for the feedback!

evgeniy-paltsev replied:
Thanks!

laurenmurphyx64 commented:
Note for people who might be coming back to this PR: After feedback from Jen offline, I dropped the commit changing the CMakeLists.txt from an explicit file list to a GLOB.

laurenmurphyx64 committed:
samples: tensorflow: change from file list to glob in cmakelists.txt

Changes CMakeLists.txt in TensorFlow samples from explicit file
list to a GLOB capturing src/* to make TensorFlow samples consistent
with other samples in tree.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>
